### ISSUE TITLE
install ubuntu 20.04

### ISSUE THREAD

**Issue by david739** [peercoin/peercoin#636]
hi, what should I do for a successful installation, thank you

sudo apt-get install build-essential libtool autotools-dev automake pkg-config bsdmainutils python3
sudo apt-get install libqt5gui5 libqt5core5a libqt5dbus5 qttools5-dev qttools5-dev-tools
sudo apt-get install libqrencode-dev

git clone https://github.com/peercoin/peercoin.git
./autogen.sh
./configure
make
make install

./autogen.sh
libtoolize: copying file 'build-aux/m4/lt~obsolete.m4'
configure.ac:88: installing 'build-aux/compile'
configure.ac:44: installing 'build-aux/missing'
src/Makefile.am: installing 'build-aux/depcomp'

./configure
checking whether to build Peercoin GUI... yes (Qt5)
checking for Berkeley DB C++ headers... no
configure: error: libdb_cxx headers missing, Peercoin requires this library for wallet functionality (--disable-wallet to disable wallet functionality)

make
error

**Comment by peerchemist** [comment]
If you really want to build it yourself, see the tutorial: https://github.com/peercoin/peercoin/blob/master/doc/build-unix.md#ubuntu--debian

**Comment by david739** [comment]
hello, we have tried everything and combined it in different ways, can you please advise me what else needs to be added to the procedure, what I have above to make it work

**Comment by peerchemist** [comment]
You clearly lack packages:

> libdb-dev and libdb++-dev packages

**Comment by david739** [comment]
hello thanks and please don't you know how to get it into ubuntu I entered


root@vmi864101:~/ppc# sudo apt-get install libdb-dev and libdb++-dev packages
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package and
E: Unable to locate package packages

**Comment by david739** [comment]
hi, I installed the package, thank you, but another error, it can't be done so that I don't have to install Berkeley DB 4.8

checking for lupdate... /usr/lib/x86_64-linux-gnu/qt5/bin/lupdate
checking whether to build Peercoin GUI... yes (Qt5)
checking for Berkeley DB C++ headers... default
configure: error: Found Berkeley DB other than 4.8, required for portable wallets (--with-incompatible-bdb to ignore or --disable-wallet to disable wallet functionality)

**Comment by david739** [comment]
so I installed Berkeley DB 4.8 did the install here is the description and errors. is it at all possible to install it on ubuntu 20.04 because it is not normal, what is the error again

make

  GEN      qt/locale/bitcoin_sv.qm
  GEN      qt/locale/bitcoin_szl.qm
Removed plural forms as the target language has less forms.
If this sounds wrong, possibly the target language is not set or recognized.
  GEN      qt/locale/bitcoin_ta.qm
  GEN      qt/locale/bitcoin_te.qm
  GEN      qt/locale/bitcoin_th.qm
  GEN      qt/locale/bitcoin_tr.qm
Warning: dropping duplicate messages in 'qt/locale/bitcoin_tr.qm'
(try -verbose for more info).
Removed plural forms as the target language has less forms.
If this sounds wrong, possibly the target language is not set or recognized.
  GEN      qt/locale/bitcoin_uk.qm
Removed plural forms as the target language has less forms.
If this sounds wrong, possibly the target language is not set or recognized.
  GEN      qt/locale/bitcoin_ur.qm
  GEN      qt/locale/bitcoin_uz@Cyrl.qm
  GEN      qt/locale/bitcoin_uz@Latn.qm
  GEN      qt/locale/bitcoin_vi.qm
Warning: dropping duplicate messages in 'qt/locale/bitcoin_vi.qm'
(try -verbose for more info).
  GEN      qt/locale/bitcoin_yo.qm
  GEN      qt/locale/bitcoin_zh-Hans.qm
  GEN      qt/locale/bitcoin_zh.qm
  GEN      qt/locale/bitcoin_zh_CN.qm
  GEN      qt/locale/bitcoin_zh_HK.qm
  GEN      qt/locale/bitcoin_zh_TW.qm
  GEN      qt/locale/bitcoin_zu.qm
  GEN      qt/qrc_bitcoin_locale.cpp
  CXX      qt/libbitcoinqt_a-qrc_bitcoin_locale.o
  AR       qt/libbitcoinqt.a
  CXXLD    qt/peercoin-qt
  CXX      support/libpeercoinconsensus_la-cleanse.lo
  CXX      crypto/libpeercoinconsensus_la-aes.lo
  CXX      crypto/libpeercoinconsensus_la-chacha_poly_aead.lo
  CXX      crypto/libpeercoinconsensus_la-chacha20.lo
  CXX      crypto/libpeercoinconsensus_la-hkdf_sha256_32.lo
  CXX      crypto/libpeercoinconsensus_la-hmac_sha256.lo
  CXX      crypto/libpeercoinconsensus_la-hmac_sha512.lo
  CXX      crypto/libpeercoinconsensus_la-poly1305.lo
  CXX      crypto/libpeercoinconsensus_la-ripemd160.lo
  CXX      crypto/libpeercoinconsensus_la-sha1.lo
  CXX      crypto/libpeercoinconsensus_la-sha256.lo
  CXX      crypto/libpeercoinconsensus_la-sha512.lo
  CXX      crypto/libpeercoinconsensus_la-siphash.lo
  CXX      crypto/libpeercoinconsensus_la-sha256_sse4.lo
  CXX      libpeercoinconsensus_la-arith_uint256.lo
  CXX      consensus/libpeercoinconsensus_la-merkle.lo
  CXX      consensus/libpeercoinconsensus_la-tx_check.lo
  CXX      libpeercoinconsensus_la-hash.lo
  CXX      libpeercoinconsensus_la-timedatadummy.lo
  CXX      primitives/libpeercoinconsensus_la-block.lo
  CXX      primitives/libpeercoinconsensus_la-transaction.lo
  CXX      libpeercoinconsensus_la-pubkey.lo
  CXX      script/libpeercoinconsensus_la-bitcoinconsensus.lo
  CXX      script/libpeercoinconsensus_la-interpreter.lo
  CXX      script/libpeercoinconsensus_la-script.lo
  CXX      script/libpeercoinconsensus_la-script_error.lo
  CXX      libpeercoinconsensus_la-uint256.lo
  CXX      util/libpeercoinconsensus_la-strencodings.lo
  
make install
root@vmi864101:~/ppc# make install
Making install in src
make[1]: Entering directory '/root/ppc/src'
make[2]: Entering directory '/root/ppc/src'
make[3]: Entering directory '/root/ppc'
make[3]: Leaving directory '/root/ppc'
make[3]: Entering directory '/root/ppc/src'
make[4]: Entering directory '/root/ppc'
make[4]: Leaving directory '/root/ppc'
 /usr/bin/mkdir -p '/usr/local/lib'
 /bin/bash ../libtool --preserve-dup-deps  --mode=install /usr/bin/install -c   libpeercoinconsensus.la '/usr/local/lib'
libtool: install: /usr/bin/install -c .libs/libpeercoinconsensus.so.0.0.0 /usr/local/lib/libpeercoinconsensus.so.0.0.0
libtool: install: (cd /usr/local/lib && { ln -s -f libpeercoinconsensus.so.0.0.0 libpeercoinconsensus.so.0 || { rm -f libpeercoinconsensus.so.0 && ln -s libpeercoinconsensus.so.0.0.0 libpeercoinconsensus.so.0; }; })
libtool: install: (cd /usr/local/lib && { ln -s -f libpeercoinconsensus.so.0.0.0 libpeercoinconsensus.so || { rm -f libpeercoinconsensus.so && ln -s libpeercoinconsensus.so.0.0.0 libpeercoinconsensus.so; }; })
libtool: install: /usr/bin/install -c .libs/libpeercoinconsensus.lai /usr/local/lib/libpeercoinconsensus.la
libtool: install: /usr/bin/install -c .libs/libpeercoinconsensus.a /usr/local/lib/libpeercoinconsensus.a
libtool: install: chmod 644 /usr/local/lib/libpeercoinconsensus.a
libtool: install: /usr/bin/ranlib /usr/local/lib/libpeercoinconsensus.a
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/sbin" ldconfig -n /usr/local/lib
----------------------------------------------------------------------
Libraries have been installed in:
   /usr/local/lib

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the '-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the 'LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the 'LD_RUN_PATH' environment variable
     during linking
   - use the '-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to '/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------
 /usr/bin/mkdir -p '/usr/local/bin'
  /bin/bash ../libtool --preserve-dup-deps  --mode=install /usr/bin/install -c peercoind peercoin-cli peercoin-tx peercoin-wallet bench/bench_bitcoin qt/peercoin-qt '/usr/local/bin'
libtool: install: /usr/bin/install -c peercoind /usr/local/bin/peercoind
libtool: install: /usr/bin/install -c peercoin-cli /usr/local/bin/peercoin-cli
libtool: install: /usr/bin/install -c peercoin-tx /usr/local/bin/peercoin-tx
libtool: install: /usr/bin/install -c peercoin-wallet /usr/local/bin/peercoin-wallet
libtool: install: /usr/bin/install -c bench/bench_bitcoin /usr/local/bin/bench_bitcoin
libtool: install: /usr/bin/install -c qt/peercoin-qt /usr/local/bin/peercoin-qt
 /usr/bin/mkdir -p '/usr/local/include'
 /usr/bin/install -c -m 644 script/bitcoinconsensus.h '/usr/local/include'
make[3]: Leaving directory '/root/ppc/src'
make[2]: Leaving directory '/root/ppc/src'
make[1]: Leaving directory '/root/ppc/src'
Making install in doc/man
make[1]: Entering directory '/root/ppc/doc/man'
make[2]: Entering directory '/root/ppc/doc/man'
make[2]: Nothing to be done for 'install-exec-am'.
 /usr/bin/mkdir -p '/usr/local/share/man/man1'
 /usr/bin/install -c -m 644 peercoind.1 peercoin-qt.1 peercoin-cli.1 peercoin-tx.1 peercoin-wallet.1 '/usr/local/share/man/man1'
make[2]: Leaving directory '/root/ppc/doc/man'
make[1]: Leaving directory '/root/ppc/doc/man'
make[1]: Entering directory '/root/ppc'
make[2]: Entering directory '/root/ppc'
make[2]: Nothing to be done for 'install-exec-am'.
 /usr/bin/mkdir -p '/usr/local/lib/pkgconfig'
 /usr/bin/install -c -m 644 libbitcoinconsensus.pc '/usr/local/lib/pkgconfig'
make[2]: Leaving directory '/root/ppc'
make[1]: Leaving directory '/root/ppc'
root@vmi864101:~/ppc#

**Comment by peerchemist** [comment]
Script is telling you what is wrong, just read.

```
BerkeleyDB is required for the wallet functionality. Historically Peercoin was first deployed with now deprecated BerkeleyDB-4.8, which has resulted in the need to upkeep the compatibility with those ancient deployments to this day. BerkeleyDB-4.8 is not compatible with more modern BerkeleyDB-5.1 and BerkeleyDB-5.3. Peercoin has inherited this in the first versions and this is why Peercoin is still officially shipped out with BerkeleyDB-4.8. However if you running a new wallet on a new installation there is absolutely no need to run old and deprecated BerkeleyDB-4.8. Just use the one avaliable in the repository of your distribution.

Ubuntu and Debian have their own libdb-dev and libdb++-dev packages, but these will install BerkeleyDB 5.1 or later. This will break binary wallet compatibility with the distributed executables, which are based on BerkeleyDB 4.8. If you do not care about wallet compatibility, pass --with-incompatible-bdb to configure.
```

Just pass `--with-incompatible-bdb` to configure. You don't need libdb4.8.
